src/views/containerNode.ts

```diff
@@ -15,23 +15,48 @@ export class ContainerNode extends DataNode {
         super(nodeData, parent);
     }
 
+    private _containerType: ContainerType;
+
     public get projectBasePath() {
         return this._project.uri && Uri.parse(this._project.uri).fsPath;
     }
 
-    public getContainerType(): string {
+    public getContainerType(): ContainerType {
+        if (this._containerType) {
+            return this._containerType;
+        }
+
         const containerPath: string = this._nodeData.path || "";
         if (containerPath.startsWith(ContainerPath.JRE)) {
-            return ContainerType.JRE;
+            this._containerType = ContainerType.JRE;
         } else if (containerPath.startsWith(ContainerPath.Maven)) {
-            return ContainerType.Maven;
+            this._containerType = ContainerType.Maven;
         } else if (containerPath.startsWith(ContainerPath.Gradle)) {
-            return ContainerType.Gradle;
+            this._containerType = ContainerType.Gradle;
         } else if (containerPath.startsWith(ContainerPath.ReferencedLibrary) && this._project.isUnmanagedFolder()) {
             // currently, we only support editing referenced libraries in unmanaged folders
-            return ContainerType.ReferencedLibrary;
+            this._containerType = ContainerType.ReferencedLibrary;
+        } else {
+            this._containerType = ContainerType.Unknown;
         }
-        return ContainerType.Unknown;
+
+        return this._containerType;
+    }
+
+    public getLabel(): string {
+        if (this._nodeData.metaData?.['maven.groupId']) {
+            return `${this._nodeData.metaData?.['maven.groupId']}:${this._nodeData.metaData?.['maven.artifactId']}:${this._nodeData.metaData?.['maven.version']}`;
```

Comment:
Just an ask:

Those metadata only exist for maven project, since they are provided by the maven project importer. Which means the dependencies of gradle project could not have the new display style. Is it as expected for you?

Reply:
That is correct, however I thought that those are the same for Gradle, but it seems they are not. It only works for Maven, as when gradle project is loaded, the jdtls is not recognizing dependencies as Gradle, instead it is.
![image](https://github.com/user-attachments/assets/489bd312-14a5-4fc3-9f73-159d3c0ad3a6)

Do you know maybe where can I look to resolve this?

Reply:
Because in JDT.LS, maven project importer and gradle project importer are two different components. So far only maven project importer leaves those metadata while the gradle project importer does not.

I've no idea how we can achieve further given the current situation. Let me think a little bit.

Reply:
Could you point me in direction where I can look at that implementation for maven and gradle importer?

Reply:
That will be a little bit complex.

This is maven importer: https://github.com/eclipse-m2e/m2e-core

For gradle importer we have two options:

- https://github.com/microsoft/build-server-for-gradle & https://github.com/microsoft/vscode-gradle/tree/develop/extension/jdtls.ext/com.microsoft.gradle.bs.importer (recommended one right now)
- https://github.com/eclipse/buildship

They are separate independent projects. I think you can check if we can do similar thing as maven at the gradle importer side. Check if we can add those metadata into classpath entries [here](https://github.com/microsoft/vscode-gradle/blob/9542e919a4b29c6ffff683c3991e3e498fdfd34a/extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerBuildSupport.java#L643)

Reply:
If you are willing to address the gradle side issue, I think we can put it into another PR/issue. And focus the maven side in this one.

Reply:
Would it be feasible to have separate display of jar's names for Maven and Gradle? I'll try to address the Gradle side as well in timely manner.

Reply:
This is exactly what I mean